test/e2e/testnet/node.go

```diff
@@ -41,7 +47,9 @@ type Node struct {
 	SignerKey      crypto.PrivKey
 	NetworkKey     crypto.PrivKey
 	SelfDelegation int64
-	Instance       *knuu.Instance
+	Instance       *instance.Instance
+	sidecars       []instance.SidecarManager
+	netShaper      *netshaper.NetShaper // a reference to the netshaper sidecar
```

Comment:
Can you explain to me what these new fields are for?

Reply:
Sure.
`sidecars` are actually knuu modules which are running in a sidecar next to the actual instance and shares the same network and infra. To have things tidy and clean, we extracted some of the knuu features as [composable modules](https://github.com/celestiaorg/knuu/tree/v0.15.2/pkg/sidecars). 
Docs are not ready yet on how to develop your own sidecar, but can have a look at the [code here](https://github.com/celestiaorg/knuu/blob/main/pkg/instance/sidecars.go).

`netShaper` is the new name for `BitTwister` which is now a sidecar for knuu.